libs/timeserieslib/DataSourceReader.h

```diff
@@ -33,35 +33,37 @@ namespace mkc_timeseries
             DateRange isDateRange, DateRange oosDateRange, bool performDownload)
     {
       setApiTimeFrameRepresentation(configTimeFrame);
-
-      boost::posix_time::time_duration zero(0, 0, 0);
-      boost::posix_time::ptime bStartDatetime(isDateRange.getFirstDate() - boost::gregorian::days(7), zero);
-      boost::posix_time::ptime bEndDatetime(oosDateRange.getLastDate() + boost::gregorian::days(7), zero);
-
-      std::string uri = buildDataFetchUri(ticker, bStartDatetime, bEndDatetime);
-      std::string filename = getFilename(ticker);
+      std::string uri = buildDataFetchUri(ticker, isDateRange.getFirstDate() - boost::gregorian::days(2), 
```

Comment:
Why are you adding and subtracting two days?

Reply:
The algorithm won't run without it. 

I meant to bring this up via email: without the (minimum) 2-day buffer an error is thrown from the ShortcutSearchAlgoBacktester.h that the number of occurrences is not equal to the backtest base size (lines 174 and 234). 

As far as I can tell, the data returned from the API covers the time range as expected; I'm not sure how the backtester and occurrences objects are using this data.